WordPress/openverse-frontend
# Add metrics middleware

## Fixes


Fixes #1256 by @sarayourfriend

## Description

'
Adding this to the frontend because it's the easiest to deploy (to search-staging) and it'll be nice to have something live to point the Grafana Agent at while I work on setting that up this week.

## Testing Instructions

Checkout the branch and run `pnpm dev`. Ensure that you cannot reach any `/metrics` endpoint. Comment out [this line](https://github.com/WordPress/openverse-frontend/blob/50f9eb6a9b98f3251c9d836a7f400abdd3c645aa/nuxt.config.ts#L244) and then repeat and confirm that you can reach the metrics endpoint. Visit some pages and confirm that the metrics endpoint records them appropriately. In particular visit some single result pages and make sure that the mapping works to unify all the individual single result pages into a single entry in the metrics listing (per search type).

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
